### ISSUE TITLE
Controller crashes upon any usage + Configs don't save at all

### ISSUE THREAD

**Issue by tokisuno** [roxgib/anki-contanki#89]
```Anki 23.12.1 (1a1d4d54)  (ao)
Python 3.9.15 Qt 6.6.1 PyQt 6.6.1
Platform: Linux-6.7.4-arch1-1-x86_64-with-glibc2.39

Traceback (most recent call last):
  File "aqt.webview", line 50, in cmd
  File "aqt.webview", line 157, in _onCmd
  File "aqt.webview", line 680, in _onBridgeCmd
  File "_aqt.hooks", line 5415, in __call__
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 146, in on_receive_message
    funcs[func](*args)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 165, in if_connected_wrapper
    func(self, *args, **kwargs)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 230, in poll
    self.do_axes_actions(state, axes)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 165, in if_connected_wrapper
    func(self, *args, **kwargs)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 346, in do_axes_actions
    value = axes[axis]
IndexError: list index out of range

===Add-ons (active)===
(add-on provided name [Add-on folder, installed at, version, is config changed])
AnkiConnect ['2055492159', 2024-01-21T17:03, 'None', '']
AwesomeTTS - Add speech to your flashcards ['1436550454', 2024-02-09T01:42, 'None', '']
Button Colours Good Again ['2494384865', 2022-09-27T06:18, 'None', '']
Contanki - Controller Support for Anki beta ['1898790263', 2024-01-30T21:50, 'v0.11.5-beta', '']
Japanese Pitch Accent ['148002038', 2023-05-07T09:01, 'None', '']
JapaneseFurigana ['678316993', 2024-02-12T09:06, 'None', '']
Mini Format Pack ['295889520', 2023-11-02T14:00, 'None', '']
More Overview Stats 21 ['738807903', 2021-07-30T15:13, 'None', '']
Review Heatmap ['1771074083', 2022-06-29T21:43, 'None', '']
Scale Images ['1312865748', 2023-11-02T02:45, 'None', '']
Searching PDF Reading  Note-Taking in Add Dialog ['1781298089', 2022-08-15T03:22, 'None', mod]
Show total time spent in main screen ['1238121460', 2020-07-19T04:36, 'None', '']

===IDs of active AnkiWeb add-ons===
1238121460 1312865748 1436550454 148002038 1771074083 1781298089 1898790263 2055492159 2494384865 295889520 678316993 738807903

===Add-ons (inactive)===
(add-on provided name [Add-on folder, installed at, version, is config changed])```

**Comment by tokisuno** [comment]
Since my controller doesn't have an analogue stick, it keeps stopping after `File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 346, in do_axes_actions
    value = axes[axis]
IndexError: list index out of range`, I think can be solved by making the 8bitdo controller function work across any dpad controller? I see in contanki.py there is already a function for if an 8bitdo is detected. I don't have one, but functionally the iBuffalo SNES gamepad is the same thing.

```Anki 23.12.1 (1a1d4d54)  (ao)
Python 3.9.15 Qt 6.6.1 PyQt 6.6.1
Platform: Linux-6.7.4-arch1-1-x86_64-with-glibc2.39

Traceback (most recent call last):
  File "aqt.webview", line 50, in cmd
  File "aqt.webview", line 157, in _onCmd
  File "aqt.webview", line 680, in _onBridgeCmd
  File "_aqt.hooks", line 5415, in __call__
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 146, in on_receive_message
    funcs[func](*args)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 165, in if_connected_wrapper
    func(self, *args, **kwargs)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 230, in poll
    self.do_axes_actions(state, axes)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 165, in if_connected_wrapper
    func(self, *args, **kwargs)
  File "/home/poto/.local/share/Anki2/addons21/1898790263/contanki.py", line 346, in do_axes_actions
    value = axes[axis]
IndexError: list index out of range

===Add-ons (active)===
(add-on provided name [Add-on folder, installed at, version, is config changed])
AnkiConnect ['2055492159', 2024-01-21T17:03, 'None', '']
AwesomeTTS - Add speech to your flashcards ['1436550454', 2024-02-09T01:42, 'None', '']
Button Colours Good Again ['2494384865', 2022-09-27T06:18, 'None', '']
Contanki - Controller Support for Anki beta ['1898790263', 2024-01-30T21:50, 'v0.11.5-beta', mod]
Japanese Pitch Accent ['148002038', 2023-05-07T09:01, 'None', '']
JapaneseFurigana ['678316993', 2024-02-12T09:06, 'None', '']
Mini Format Pack ['295889520', 2023-11-02T14:00, 'None', '']
More Overview Stats 21 ['738807903', 2021-07-30T15:13, 'None', '']
Review Heatmap ['1771074083', 2022-06-29T21:43, 'None', '']
Scale Images ['1312865748', 2023-11-02T02:45, 'None', '']
Searching PDF Reading  Note-Taking in Add Dialog ['1781298089', 2022-08-15T03:22, 'None', mod]
Show total time spent in main screen ['1238121460', 2020-07-19T04:36, 'None', '']

===IDs of active AnkiWeb add-ons===
1238121460 1312865748 1436550454 148002038 1771074083 1781298089 1898790263 2055492159 2494384865 295889520 678316993 738807903

===Add-ons (inactive)===
(add-on provided name [Add-on folder, installed at, version, is config changed])```

**Comment by roxgib** [comment]
Thanks for reporting, should be fixed with [1969917](https://github.com/roxgib/anki-contanki/commit/19699178fddad1e5bf8d13cbefd023349b57e0d0)

**Comment by tokisuno** [comment]
So does this mean I have to wait for a new release for this to be fixed? Or does anki automatically update addons through git?
